src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,573 @@
+   *      frequency-id 1    => 118.005 kHz
+   *      frequency-id 3040 => 137.000 kHz
+   * 
+   * @param frequency frequency in kHz
+   * @return uint16_t frequency-id
+   */
+  uint16_t FrequencyToAR62xxId(RadioFrequency frequency);
+
+  /**
+   * @brief converts an AR62xx-ID to a real frequency in kHz
+   * 
+   * @param frequency_id frequency-id
+   * @return RadioFrequency frequency in kHz
+   */
+  RadioFrequency AR62xxIdToFrequency(uint16_t frequency_id);
+
+  /**
+   * @brief generates a command to set a station in the AR62xx
+   * 
+   * @param command command-output to set the station
+   * @param station_id the id of the station which should be set
+   *        1 = active station of AR62xx
+   *        0 = passive station of AR62xx
+   * @param frequency frequency in kHz to set
+   * @param station name of the station to set
+   *        (unused)
+   * @return int command-length
+   */
+  int GenerateSetStationCommand(uint8_t *command, int station_id, RadioFrequency frequency, const TCHAR *station);
+
+  /**
+   * @brief parses the received data
+   * this is needed to save the active and passive station
+   * 
+   * @param string recived data
+   * @param len length of data
+   * @return true if data could be parsed
+   * @return false if data could not be parsed
+   */
+  bool ParseReceivedData(const char *string, size_t len);
+
+  /**
+   * @brief save frequencies
+   * 
+   * @param sz_command command which contains the frequencies
+   * @param len length of the command
+   * @param crc CRC
+   */
+  void SaveAR62xxFrequencies(uint8_t *sz_command, int len, uint16_t crc);
+
+  /**
+   * @brief Get the n-th Byte of an unsigned integer
+   * 
+   * @param number unsigned integer
+   * @param n byte-number
+   * @return uint8_t byte
+   */
+  uint8_t GetNthByte(unsigned number, int n);
+
+  /**
+   * @brief Converts to uint8_t-bytes to an unsigned integer
+   * 
+   * @param first first byte
+   * @param second second byte
+   * @return unsigned unsigned integer
+   */
+  unsigned BytesToUnsigned(uint8_t first, uint8_t second);
+
+public:
+  virtual bool PutActiveFrequency(RadioFrequency frequency,
+                                  const TCHAR *name,
+                                  OperationEnvironment &env) override;
+  virtual bool PutStandbyFrequency(RadioFrequency frequency,
+                                   const TCHAR *name,
+                                   OperationEnvironment &env) override;
+
+  /**
+   * @brief checks if data is received
+   * 
+   * @param _data data which is received
+   * @param length length of the data
+   * @param info nmea info
+   * @return true data has been received
+   * @return false no data has been received
+   */
+  virtual bool DataReceived(const void *data, size_t length, struct NMEAInfo &info) override;
+};
+
+/**
+ * @brief Construct a new AR62xxDevice::AR62xxDevice object
+ * 
+ * @param _port the port to connect to
+ */
+AR62xxDevice::AR62xxDevice(Port &_port) : port(_port) {}
```

Comment:
One-liners like this one should be inlined.

Reply:
done